streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -554,7 +556,12 @@ public void close() {
         @Override
         public void toggleDbForBulkLoading() {
             try {
-                db.compactRange(columnFamily, true, 1, 0);
+                final CompactRangeOptions crOptions = new CompactRangeOptions();
```

Comment:
`CompactRangeOptions extends RocksObject`, we need to explicitly `close()` it otherwise it's memory leak.

Since all such objects are created the same, how about just keeping a single copy of it and close it during `store.close()` similar to other xxOptions?

Reply:
Good catch. I found that the BloomFilter also needs to be closed (and isn't) -- I'll check around for more leaking Rocks objects just in case. If I find any should they go in a separate PR for fixing up older versions?

Reply:
Let's wait until the KIP is merged before this one.

Reply:
Actually we got the BloomFilter fix into the 2.2.1 release :)

Reply:
Ah right, I think then it is not blocked on anything (the KIP was semi-orthogonal to this).